### PR TITLE
Add xrefcheck

### PR DESCRIPTION
**Pull request by gromakovsky**
## Description

Use the [xrefcheck](https://github.com/serokell/xrefcheck/) tool to ensure that cross-references are valid.

## Related issues(s)

None

## ✓ Checklist for your Pull Request

Ideally a PR has all of the checkmarks set.

If something in this list is irrelevant to your PR, you should still set this
checkmark indicating that you are sure it is dealt with (be that by irrelevance).

- [x] I made sure my PR addresses a single concern, or multiple concerns which
      are inextricably linked. Otherwise I should open multiple PR's.
- [x] If your PR fixes/relates to an open issue then the description should
      reference this issue. See also [auto linking on
      github](https://help.github.com/articles/autolinked-references-and-urls/).

#### Related changes (conditional)

- Tests

  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation

  I checked whether I should update the docs and did so if necessary:

  - [x] [README](/README.md)
  - [x] Haddock

- Record your changes

  - [x] I added an entry to the [changelog](/CHANGES.md) if my changes are visible to the users
        and
  - [x] provided a migration guide for breaking changes if possible

#### Stylistic guide (mandatory)

- [x] My commit history is clean (only contains changes relating to my
      issue/pull request and no reverted-my-earlier-commit changes) and commit
      messages start with identifiers of related issues in square brackets.

  **Example:** `[#42] Short commit description`

  If necessary both of these can be achieved even after the commits have been
  made/pushed using [rebase and squash](https://git-scm.com/docs/git-rebase).

### CODE REVIEW

**Comment by gromakovsky** [comment]
Ah, I guess I should make this check optional. I am considering switching CI to GitHub actions, so maybe I'll do that first.

File: .travis.yml [hunk 1]
```diff
@@ -69,7 +69,11 @@ install:
 
 script:
 
+  # Verify cross references
+  - wget -O - https://github.com/serokell/xrefcheck/releases/download/v0.1.1/release.tar.gz | tar xzv && xrefcheck/bin/xrefcheck --ignored xrefcheck
```
Comment (Martoon-00) [inline .travis.yml:73]: Shouldn't `wget` + `tar` part be part of install section? (and maybe cached)

**Comment by Martoon-00** [comment]
Mda, 6 jobs each resolving all refs to issues located in `CHANGES.md`. 429 guaranteed, need to do something with this soon.

Probably it's worth enabling retry:
https://docs.travis-ci.com/user/common-build-problems/#travis_retry

**Comment by gromakovsky** [comment]
Doing it in 6 jobs definitely does not make sense. I'd like to try GitHub actions first before doing it properly.

**Comment by Martoon-00** [comment]
Ah right, I see

**Comment by gromakovsky** [comment]
I might end up doing `xrefcheck-action` and using it here before considering this PR done.

**Comment by gromakovsky** [comment]
Alright, I think all issues are resolved and it works pretty nicely now.

**Comment by gromakovsky** [comment]
I've also bumped the version to 1.7.0 because I want to make a new release and don't want to make another PR for that.

**Comment by Martoon-00** [comment]
Since you are going to make a release, may I quickly add one small PR? Wanna have it soon

**Comment by Martoon-00** [comment]
I was wrong, that issue is already resolved. Let's make a release :)

**Comment by gromakovsky** [comment]
Great!

**Pull request merged**